lib/auth/tls_test.go

```diff
@@ -2428,12 +2428,13 @@ func TestClusterConfigContext(t *testing.T) {
 	_, pub, err := testauthority.New().GenerateKeyPair()
 	require.NoError(t, err)
 
-	// try and generate a host cert, this should fail because we are recording
-	// at the nodes not at the proxy
+	// try and generate a host cert, this should succeed because although
+	// we are recording at the nodes not at the proxy, the proxy may
+	// need to record if a client wants to connect to an agentless node
 	_, err = proxy.GenerateHostCert(ctx, pub,
 		"a", "b", nil,
 		"localhost", types.RoleProxy, 0)
-	require.True(t, trace.IsAccessDenied(err))
+	require.NoError(t, err)
```

Comment:
Why has the error here changed?

Reply:
I changed the comment above to answer that: https://github.com/gravitational/teleport/pull/21953/files#diff-74753e67ac21ad6bd83896d46e43257063e62a5d903db9085b992fdc8c7e060fR2431-R2433